docs/content/develop/run-tests.md

```diff
@@ -0,0 +1,344 @@
+
+    ```bash
+    make testacc TEST=./google-beta TESTARGS='-run=TestAccContainerNodePool'
+    ```
+
+
+
+1. Optional: Save verbose test output to a file for analysis.
+
+    ```bash
+    TF_LOG=TRACE make testacc TEST=./google-beta TESTARGS='-run=TestAccContainerNodePool_basic' > output.log
+    ```
+
+1. Optional: Debug tests with [Delve](https://github.com/go-delve/delve). See [`dlv test` documentation](https://github.com/go-delve/delve/blob/master/Documentation/usage/dlv_test.md) for information about available flags.
+
+    ```bash
+    cd google-beta
+    TF_ACC=1 dlv test -- --test.v --test.run TestAccComputeRegionBackendService_withCdnPolicy
+    ```
+
+{{< /tab >}}
+
+{{< /tabs >}}
+
+## Test with different `terraform` versions
+
+Tests will use whatever version of the `terraform` binary is found on your `PATH`. If you are testing a change that you know only impacts certain `terraform` versions, follow these steps:
+
+1. Install [`tfenv`](https://github.com/tfutils/tfenv).
+
+1. Install the version of `terraform` you want to test.
+
+    ```bash
+    tfenv install VERSION
+    ```
+
+    Replace `VERSION` with the version you want to test.
+
+1. Run automated tests following the earlier section.
+
+1. Run manual tests following the earlier section.
+
+## Test manually
+
+Sometimes, for example for manual testing, you may want to build the provider from source and use it with `terraform`.
+
+### Build provider
+
+```bash
+## ga provider
+cd $GOPATH/src/github.com/hashicorp/terraform-provider-google
+make build
+
+## beta provider
+cd $GOPATH/src/github.com/hashicorp/terraform-provider-google-beta
+make build
+```
+
+{{< tabs "built-provider" >}}
+
+{{< tab "Developer overrides" >}}
+
+In the sections below we describe how to create a Terraform CLI configuration file, and how to make the CLI use the file via an environment variable.
+
+### Create developer overrides file
+
+Choose your architecture below.
+
+{{< tabs "setup" >}}
+
+{{< tab "Mac (ARM64 and AMD64), Linux AMD64" >}}
+
+1. Find the location where built provider binaries are created. To do this, run this command and make a note of the path value:
+
+    ```bash
+    go env GOBIN
+
+    ## If the above returns nothing, then run the command below and add "/bin" to the end of the output path.
+    go env GOPATH
+    ```
+
+1. Create an empty configuration file.
+
+    ```bash
+    ## create an empty file
+    touch ~/tf-dev-override.tfrc
+
+    ## open the file with a text editor of your choice, e.g:
+    vi ~/tf-dev-override.tfrc
+    ```
+
+    Open the empty file with a text editor and paste in these contents:
+
+    ```hcl
+    provider_installation {
+
+      # Developer overrides will stop Terraform from downloading the listed
+      # providers their origin provider registries.
+      dev_overrides {
+          "hashicorp/google" = "<REPLACE-ME>/bin"
```

Comment:
see https://developers.google.com/style/placeholders for style guidance - should be a meaningful value, all caps, with underscores.